tools/querytee/response_comparator.go

```diff
@@ -391,10 +430,36 @@ func compareVector(expectedRaw, actualRaw json.RawMessage, queryEvaluationTime t
 		return err
 	}
 
-	if allVectorSamplesWithinRecentSampleWindow(expected, queryEvaluationTime, opts) && allVectorSamplesWithinRecentSampleWindow(actual, queryEvaluationTime, opts) {
+	if allVectorSamplesOutsideComparableWindow(expected, queryEvaluationTime, opts) && allVectorSamplesOutsideComparableWindow(actual, queryEvaluationTime, opts) {
 		return nil
 	}
 
+	// Warning: filtering samples can give out confusing error messages. For example if the actual response had
+	// matching series, but all sample timestamps were before SkipSamplesBefore, while expected response had samples
+	// after SkipSamplesBefore: instead of saying mismatch, it will instead say that some metrics is missing, because
+	// we filter them here.
+	eOrgLen, aOrgLen := len(expected), len(actual)
+	expected = filterSlice(expected, func(p *model.Sample) bool {
+		return p.Timestamp < opts.SkipSamplesBefore
+	})
+	actual = filterSlice(actual, func(p *model.Sample) bool {
+		return p.Timestamp < opts.SkipSamplesBefore
+	})
+	eChanged, aChanged := len(expected) != eOrgLen, len(actual) != aOrgLen
+	defer func() {
+		if retErr != nil {
+			warning := ""
+			if eChanged && aChanged {
+				warning = " (also, some samples were filtered out from the expected and actual response due to the 'skip samples before')"
```

Comment:
[nit] I would being even more explicit. Same comment applies to other warning messages:

```suggestion
				warning = " (some samples were filtered out from the expected and actual response due to the 'skip samples before'; if all samples have been filtered out, this could cause the check on the expected number of metrics to fail)"
```